File: lib/android/app/src/main/java/com/reactnativenavigation/options/params/Colour.java
package com.reactnativenavigation.options.params;

import android.graphics.Color;
import androidx.annotation.ColorInt;
import androidx.annotation.NonNull;

public class Colour extends Param<Integer>{

    public Colour(@ColorInt int color) {
        super(color);
    }

    @NonNull
    @SuppressWarnings("MagicNumber")
    @Override
    public String toString() {
        return String.format("#%06X", (0xFFFFFF & get()));
    }

    public boolean hasTransparency() {
        return hasValue() && Color.alpha(value) < 1;
    }
}

File: lib/android/app/src/main/java/com/reactnativenavigation/options/params/Colour.kt
package com.reactnativenavigation.options.params

import android.graphics.Color
import androidx.annotation.ColorInt
open class Colour(@ColorInt color: Int) : Param<Int>(color) {
    override fun toString(): String {
        return String.format("#%06X", 0xFFFFFF and get())
    }

    fun hasTransparency(): Boolean {
        return hasValue() && Color.alpha(value) < 1
    }
}


object NullColor : Colour(0) {
    override fun hasValue(): Boolean {
        return false
    }

    override fun toString(): String {
        return "Null Color"
    }
}